crates/ruff_python_formatter/src/string/normalize.rs

```diff
@@ -53,32 +55,36 @@ impl<'a, 'src> StringNormalizer<'a, 'src> {
             // The reason to preserve the quotes is based on the assumption that
             // the original f-string is valid in terms of quoting, and we don't
             // want to change that to make it invalid.
-            if (context.f_string().flags().is_triple_quoted() && !string.flags().is_triple_quoted())
-                || self.context.options().target_version().supports_pep_701()
-            {
-                self.quoting
-            } else {
-                Quoting::Preserve
+            (Quoting::CanChange, FStringState::InsideExpressionElement(context)) => {
+                if (context.f_string().flags().is_triple_quoted()
+                    && !string.flags().is_triple_quoted())
+                    || self.context.options().target_version().supports_pep_701()
+                {
+                    Quoting::CanChange
+                } else {
+                    Quoting::Preserve
+                }
             }
-        } else {
-            self.quoting
+
+            (Quoting::CanChange, _) => Quoting::CanChange,
         }
```

Comment:
I'm a bit confused with this diff so just want to make sure it's correct. In the earlier version, the quoting could be anything but if the f-string state is inside an expression element, then it returns the quoting is the condition matches other `Preserve`. In the new version, I see the f-string state is combined with only `CanChange`, does it also cover the `Preserve`?

Reply:
This should be the same because if the option was `Preserve` in the old code then:

* It returned `Preserve` in the else branch (because `self.quoting` is preserve)
* It returned `Preserve` in the f-string branch because `self.quoting` is preserve)
* It returned `Preserve` in the `if...else` branch because it is hardcoded to `Quoating::Preserve`